apps/client/src/app/core/data/lazy-data.service.ts

```diff
@@ -146,27 +149,36 @@ export class LazyDataService {
       }),
       filter((recipes) => recipes !== undefined),
       map((recipes) => {
-        return recipes.find((r) => r.id.toString() === id.toString()) || this.data.recipes.find((r) => r.id.toString() === id.toString());
+        return (
+          ((recipes as unknown) as Craft[]).find((r) => r.id.toString() === id.toString()) ||
+          ((this.data.recipes as unknown) as Craft[]).find((r) => r.id.toString() === id.toString())
```

Comment:
Why do we have to cast as unknown before casting as `Craft[]`? The recipe model and the craft one should match, if they don't, the Craft model needs to be readjusted because it's used by simulator where we feed recipes ^^

Reply:
Yep, you are correct, they don't match up quite right. Craft actually comes from the sim, so I can't fix it here.

But that makes these auto-generated types really valuable for validating our assumptions about the shape of this json data. Something we could do to fix it here is to just write an adapter from lazyrecipe->craft, that way we're not breaking everything that already relies on craft.